Client/Frontend/Home/HistoryHighlights/HistoryHighlightsManager.swift

```diff
@@ -115,7 +120,7 @@ class HistoryHighlightsManager: HistoryHighlightsManagerProtocol {
                 !tabs.contains { highlights.urlFromString == $0.lastKnownUrl }
             }
 
-            filterHighlights = SponsoredContentFilterUtility().filterSponsoredHighlights(from: filterHighlights)
+            filterHighlights = SponsoredContentFilterUtility().filterSponsoredHighlights(from: shouldFilterLastURL ? filterHighlights : highlights)
```

Comment:
We should always use `filterHighlights` because the we want to filter out the opened tabs. So this change is causing to include open tabs in Recently visited. I think the issue for this bug might be related to a missing reload case when the home button is pressed

Reply:
Hey @yoanarios thank you for you explanation. I did some more deep research and found that we should always filter highlights, as you told me, for the case when the url is opened in a tab. So I reverted my first changes. When the home button is press the home view is already reloaded, but at the moment after the home tab replaces the last url visited we should reload the data since at that point we are sure that the last url different from homepage is included in the filtered highlights. So I added this change.
I attach a video of how it is working now. Also all the tests pass with this change so no updates to the tests. Please let me know what you think about this solution. Also works properly for the case with multiple tabs opened, as you can see in the movie. Thank you!

https://user-images.githubusercontent.com/46751540/219051387-d5a91fef-a7e6-4a71-8424-5c3e1ab5a842.mov

Reply:
> 

Thank you, I will review soon